src/shogun2-util/src/main/java/de/terrestris/shogun2/util/interceptor/CachedServletInputStream.java

```diff
@@ -0,0 +1,63 @@
+package de.terrestris.shogun2.util.interceptor;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import javax.servlet.ReadListener;
+import javax.servlet.ServletInputStream;
+
+/**
+ *
+ * An inputstream which reads the cached request body and has mutable
+ * request URI and params.
+ *
```

Comment:
Please ad another `@see` pointing to http://stackoverflow.com/questions/10210645/http-servlet-request-lose-params-from-post-body-after-read-it-once here.